src/main/java/org/jitsi/jicofo/Participant.java

```diff
@@ -454,6 +481,30 @@ public boolean isSessionEstablished()
         return jingleSession != null;
     }
 
+    /**
+     * Terminates the current {@code BridgeSession}, terminates the channel
+     * allocator and resets any fields related to the session.
+     *
+     * @return {@code BridgeSession} from which this {@code Participant} has
+     * been removed or {@code null} if this {@link Participant} was not part
+     * of any bridge session.
+     */
+    JitsiMeetConferenceImpl.BridgeSession terminateBridgeSession()
+    {
+        JitsiMeetConferenceImpl.BridgeSession _session = this.bridgeSession;
```

Comment:
Both BridgeSession and ChannelAllocator are set in one place currently. I'm thinking of BridgeSession member field as an indicator that participant has been assigned to a BridgeSession. ChannelAllocator is something that makes it join the session. Now the terminateBridgeSession method is used to remove the Participant from the session. I can't think of any use case in which you would want to clear the session, but not the channel allocator (or vice versa). Given the above the old code you mention does not make sense to me.

I've been thinking of making a method which creates the ChannelAllocator inside of Participant to not allow a situation when BridgeSession is changed without the ChannelAllocator being changed at the same time or vice versa. It could be named "joinBridgeSession" and take BridgeSession as the argument and create a ChannelAllocator inside. Then setBridgeSession and setChannelAllocator would be removed.